### ISSUE TITLE
Rule suggestion: `vue/require-explicit-slots` to require all slots to be documented

### ISSUE THREAD

**Issue by Amixx** [vuejs/eslint-plugin-vue#2294]
**Please describe what the rule should do:**
I would like a rule that requires `defineSlots` to be present if there are any slots in the component - could be called `vue/no-undeclated-slots` or something similar. I did not find a rule that does this in the rule list, and did not find a relevant issue.

**What category should the rule belong to?**
[X] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<template>
<slot></slot>
</template>

<script setup lang="ts">
// any code, except `defineSlots`
</script>
```

**Additional context**
I am sorry if this issue is not submitted correctly, and will fix any errors. I am also willing to try to implement the rule myself if anyone can give me a starting point (maybe a rule that is somewhat similar)

**Comment by FloEdelmann** [comment]
Sounds good!

For reference, here is the documentation of `defineSlots`: https://vuejs.org/api/sfc-script-setup.html#defineslots
That should also be included in the rule documentation.

Similar rules:
* https://eslint.vuejs.org/rules/require-explicit-emits.html
* https://eslint.vuejs.org/rules/no-undef-components.html

As a name for the new rule, I'd suggest `vue/require-explicit-slots` to be consistent with `vue/require-explicit-emits`.

**Comment by Amixx** [comment]
Thanks for the quick comment, I will try to implement it over the weekend.

**Comment by mussinbenarbia** [comment]
@Amixx @FloEdelmann I noticed that there was no PR for this yet so I gave this a try [here](https://github.com/vuejs/eslint-plugin-vue/pull/2325) 🙇 I hope it's ok
